internal/bminventory/inventory_test.go

```diff
@@ -13188,6 +13178,65 @@ var _ = Describe("TestRegisterCluster", func() {
 		}
 	})
 
+	Context("Cluster Mirror Registry", func() {
+		const (
+			mirrorRegistryCertificate = "-----BEGIN CERTIFICATE-----\n    certificate contents\n-----END CERTIFICATE------"
+			sourceRegistry            = "quay.io"
+			mirrorRegistry            = "example-user-registry.com"
+		)
+
+		getSecureRegistryToml := func() string {
+			return fmt.Sprintf(`
+[[registry]]
+location = "%s"
+
+[[registry.mirror]]
+location = "%s"
+`,
+				sourceRegistry,
+				mirrorRegistry,
+			)
+		}
+
+		getClusterCreateParams := func() *models.ClusterCreateParams {
+			return &models.ClusterCreateParams{
+				Name:                 swag.String("some-cluster-name"),
+				OpenshiftVersion:     swag.String(common.TestDefaultConfig.OpenShiftVersion),
+				PullSecret:           swag.String(fakePullSecret),
+				CPUArchitecture:      models.ClusterCPUArchitectureX8664,
+				HighAvailabilityMode: swag.String(models.ClusterCreateParamsHighAvailabilityModeFull),
+			}
+		}
+
+		getMirrorRegistryConfigurations := func(registriesToml, certificate string) (*v1beta1.MirrorRegistryConfiguration, []configv1.ImageDigestMirrors) {
+			imageDigestMirrors, imageTagMirrors, insecure, err := mirrorregistries.GetImageRegistries(registriesToml)
+			Expect(err).To(Not(HaveOccurred()))
+
+			mirrors := &v1beta1.MirrorRegistryConfiguration{
+				ImageDigestMirrors: imageDigestMirrors,
+				ImageTagMirrors:    imageTagMirrors,
+				Insecure:           insecure,
+				CaBundleCrt:        certificate,
+				RegistriesConf:     registriesToml,
+			}
+
+			return mirrors, imageDigestMirrors
+		}
+
+		It("Validate mirror registry saved on DB", func() {
+			mockClusterRegisterSuccess(true)
+			mockAMSSubscription(ctx)
+			conf, _ := getMirrorRegistryConfigurations(getSecureRegistryToml(), mirrorRegistryCertificate)
+			params := getClusterCreateParams()
+			c, err := bm.RegisterClusterInternal(ctx, nil, conf, installer.V2RegisterClusterParams{NewClusterParams: params})
+			Expect(err).ShouldNot(HaveOccurred())
+
+			var clusterObj common.Cluster
+			Expect(db.First(&clusterObj, "id = ?", c.ID).Error).ShouldNot(HaveOccurred())
+			Expect(clusterObj.MirrorRegistryConfiguration).ShouldNot(BeNil())
```

Comment:
Is there anything about this content you can verify?

Reply:
Sure, added verifications